### PR TITLE
Loop limit

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes https://github.com/leonardoalt/yools/issues/16

### CODE REVIEW

File: tests/integration.rs [hunk 1]
```diff
@@ -85,6 +85,25 @@ fn test_syntax_assert_pass_update() {
     );
 }
 
+fn loop_unroll(source: &str) -> Option<u64> {
+    let v: Vec<_> = source.split('\n').collect();
```
Comment (chriseth) [inline tests/integration.rs:89]: Can't you directly loop over the iterator without collecting it into a vector?

File: tests/assert_pass/for_loop_over.yul [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+	let max := calldataload(0)
+	if gt(max, 10) { revert(0, 0) }
+	let x
+	let i
+	for {} lt(i, max) {
+		x := add(x, 1)
```
Comment (chriseth) [inline tests/assert_pass/for_loop_over.yul:7]: I mean it doesn't matter, but for the sake of readability, wouldn't you rather swap the post and the body part of this loop?

File: tests/integration.rs (hunk 1 above)
Comment (leonardoalt) [reply tests/integration.rs:89]: yea, I'll rewrite this function, Thibaut gave me some tips

File: tests/assert_pass/for_loop_over.yul (hunk 2 above)
Comment (leonardoalt) [reply tests/assert_pass/for_loop_over.yul:7]: wait, I thought this was the body, is it not?

Comment (chriseth) [reply tests/assert_pass/for_loop_over.yul:7]: pre condition post body, just like in a c-style for loop

Comment (leonardoalt) [reply tests/assert_pass/for_loop_over.yul:7]: yea, I've written a bunch of tests wrong then haha but doesn't really affect the semantics

File: tests/integration.rs [hunk 3]
```diff
@@ -80,6 +80,17 @@ fn test_syntax_assert_pass_update() {
     );
 }
 
+fn loop_unroll(source: &str) -> Option<u64> {
+    source.split('\n').find_map(|l| {
+        l.starts_with("// loop_unroll:")
+            .then(|| l.split(':').nth(1).unwrap().parse::<u64>().unwrap())
+    })
```
Comment (leonardoalt) [inline tests/integration.rs:87]: @chriseth better? ;)

**Review by chriseth — APPROVED** [review]

**Pull request merged**